File: Trees/RightView.java
package Trees;
import java.util.LinkedList;
import java.util.Queue;

import Trees.BinaryTree.*;
public class RightView {
	public static void main(String[] args) {
		BinaryTree tree = new BinaryTree();
		tree.root = new TreeNode(12);
		tree.root.left = new TreeNode(10);
		tree.root.right = new TreeNode(30);
		tree.root.right.left = new TreeNode(25);
		tree.root.right.right = new TreeNode(40);
		printRight(tree.root);
	}
	static void printRight(TreeNode root){
		if(root==null) return;
		Queue<TreeNode> queue = new LinkedList<TreeNode>();
		queue.add(root);
		queue.add(null);
		while(!queue.isEmpty()){
			TreeNode node = queue.remove();
			if(node==null && !queue.isEmpty()){
				queue.add(null);
			}
			if(node!=null && node.left!=null){
				queue.add(node.left);
			}
			if(node!=null && node.right!=null){
				queue.add(node.right);
			}
			if(node!=null && queue.peek()==null){
				System.out.print(node.val+" ");
			}
			
		}
			
	}
}

File: C++/Trees/RightView.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long
struct node
{
    int data;
    node *left;
    node *right;

    node(int val)
    {
        data = val;
        left = NULL;
        right = NULL;
    }
};

void RightView(node *root)
{
    if (root == NULL)
    {
        return;
    }

    queue<node *> q;
    q.push(root);
    q.push(NULL);
    int temp;

    while (!q.empty())
    {
        node *node = q.front();
        q.pop();
        if (node != NULL)
        {
            temp = node->data;
            if (node->left)
                q.push(node->left);
            if (node->right)
                q.push(node->right);
        }
        else if (!q.empty())
        {
            cout << temp << " ";
            q.push(NULL);
        }
    }
    cout << temp;
}

void rightview(node *root)
{
    if (root == NULL)
    {
        return;
    }

    queue<node *> q;
    q.push(root);
    // q.push(NULL);

    while (!q.empty())
    {
        int n = q.size();

        for (int i = 0; i < n; i++)
        {
            node *node = q.front();
            q.pop();
            if (i == (n - 1))
            {
                cout << node->data << " ";
            }

            if (node->left != NULL)
                q.push(node->left);
            if (node->right != NULL)
                q.push(node->right);
        }
    }
}

main()
{
    node *root = new node(1);
    root->left = new node(2);
    root->left->left = new node(3);
    root->left->right = new node(4);
    root->left->right->left = new node(6);
    root->right = new node(5);

    RightView(root);
}